src/eins-hwinfo.c

```diff
@@ -135,129 +156,18 @@ eins_hwinfo_get_disk_space_for_partition (GFile   *file,
-                                      g_steal_pointer (&payload));
-}
-
-static gboolean
-record_disk_space (gpointer unused)
+eins_hwinfo_get_space_for_rootfs (DiskSpaceType *diskspace)
 {
   g_autoptr(GFile) root = g_file_new_for_path ("/");
-
-  record_disk_space_for (root, SYSROOT_DISK_SPACE_EVENT);
-
-  return G_SOURCE_CONTINUE;
-}
-
-static void
-start_recording_disk_space (void)
-{
-  record_disk_space (NULL);
-  g_timeout_add_seconds (RECORD_DISK_SPACE_INTERVAL_SECONDS, record_disk_space,
-                         NULL);
-}
-
-static void
-boot_finished_cb (GFileMonitor     *monitor,
-                  GFile            *booted,
-                  GFile            *other_file,
-                  GFileMonitorEvent event_type,
-                  gpointer          user_data)
-{
-  /* Any event will do */
-  g_debug ("got (GFileMonitorEvent) %d for %s", event_type, BOOTED_FLAG_FILE_PATH);
-  start_recording_disk_space ();
-
-  g_signal_handlers_disconnect_by_func (monitor, boot_finished_cb, NULL);
-  g_object_unref (monitor);
-}
-
-/* On the first boot, the root partition is extended to fill the disk, in the
- * background. We may be running before this process has completed; in that
- * case, we need to wait. Rather than monitoring eos-firstboot.service via
- * systemd's D-Bus API, we look for a flag file in /var.
- */
-static gboolean
-start_recording_disk_space_when_booted (gpointer data)
-{
-  g_autoptr(GFile) booted = g_file_new_for_path (BOOTED_FLAG_FILE_PATH);
-  g_autoptr(GFileMonitor) monitor = NULL;
   g_autoptr(GError) error = NULL;
 
-  if (g_file_query_exists (booted, NULL))
-    {
-      g_debug ("%s already exists", BOOTED_FLAG_FILE_PATH);
-      start_recording_disk_space ();
-    }
-  else if (!(monitor = g_file_monitor_file (booted, G_FILE_MONITOR_NONE,
-                                            NULL, &error)))
-    {
-      g_warning ("Couldn't watch %s: %s",
-                 BOOTED_FLAG_FILE_PATH, error->message);
-      start_recording_disk_space ();
-    }
-  else
-    {
-      g_debug ("Waiting for %s to appear before reporting disk space",
-               BOOTED_FLAG_FILE_PATH);
-
-      /* Ownership is transferred to boot_finished_cb() */
-      g_signal_connect (g_steal_pointer (&monitor), "changed",
-                        (GCallback) boot_finished_cb,
-                        NULL);
-    }
-
-  return G_SOURCE_REMOVE;
-}
-
-GVariant *
-eins_hwinfo_get_ram_size (void)
-{
-  glibtop_mem mem;
-  guint32 size_mib;
-
-  glibtop_get_mem (&mem);
-  size_mib = round_to_nearest (mem.total, ONE_MIB_IN_BYTES);
-  return g_variant_new_uint32 (size_mib);
-}
-
-static gboolean
-record_ram_size (gpointer unused)
-{
-  emtr_event_recorder_record_event (emtr_event_recorder_get_default (),
-                                    RAM_SIZE_EVENT,
-                                    eins_hwinfo_get_ram_size ());
-  return G_SOURCE_REMOVE;
+  eins_hwinfo_get_disk_space_for_partition (root, diskspace, &error);
```

Comment:
If you are just going to ignore the error you can just pass `NULL`. But the caller should log the error with `g_warning()` as it previously did.

Reply:
Fixed in the other way as mentioned above.